script.py

```diff
@@ -0,0 +1,41 @@
+import requests
+import pprint
+import config
+
+def main():
+    token = generateToken(config.client_id, config.client_secret)
+    if token:
+        response = requestVehicles(token)
+        printInfo(response)
+    return 0
+
+def generateToken(client_id, client_secret):
+    tokenResponse = requests.post("https://api.oregonstate.edu/oauth2/token", data={
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "grant_type": "client_credentials"
+    })
+    if(tokenResponse.status_code == 200):
+        return tokenResponse.json()['access_token']
+    else:
+        print("Invalid credentials")
+        return None
+
+def requestVehicles(token):
+    parameters = {
+        "Authorization": f"Bearer {token}"
+    }
+    return requests.get("https://api.oregonstate.edu/v1/beaverbus/vehicles", headers=parameters)
+    
+def printInfo(response):
+    pp = pprint.PrettyPrinter()
+    
+    print(response.json()["data"][0]["attributes"]["name"], "Stats:")
+    print("Latitude:", response.json()["data"][0]["attributes"]["latitude"])
+    print("Longitude:", response.json()["data"][0]["attributes"]["longitude"])
+    print("Current Speed:", response.json()["data"][0]["attributes"]["speed"])
```

Comment:
So right now this script has the function to fetch all vehicles from the API, however, in your `printInfo` method, you only fetch the first vehicles (`response.json()["data"][0]`) from the list. Can you modify your code here to allow printing all vehicles information?

Reply:
In addition, `response.json()["data"]` was actually multiple time. You should store it into a variable to make it reusable. Also, try to use `f-string` to format your strings instead of concatenating them with `,`.

Ref: https://realpython.com/python-f-strings/#multiline-f-strings